Correction to uninitialized constant Thin::Server::Logging

Ginkggo opened the pull request:
This should fix https://github.com/macournoyer/thin/issues/447

Ginkggo committed:
added requires for missing logger

ioquatix commented:
Can you please review the test failures? Thanks!

Commit:
logic for handling rack versions

ioquatix commented:
This looks okay to me, but I'll review it in more detail after you remove the white space changes.

Commit:
undo whitespace removal

ioquatix commented:
Okay, I reviewed the change. I'm not sure I understand how this fix is supposed to work.

I think you need to modify https://github.com/macournoyer/thin/blob/master/lib/thin/server.rb and add `require_relative "logging"`.

Ginkggo commented:
@ioquatix That was my first commit, I did `require 'thin' ` because `daemonizable` was also being uninitialized
I tested that change and it worked but I didn't understand why the tests were failing because of it.
I added the rack version handling since that was what the tests were failing to find and added the logging in there as well to ensure reference.

tbh I could have left it in server.rb

ioquatix commented:
Hmm, maybe CI is broken, let me check.

jaevans commented:
Any progress on this with either CI or this fix? We're having the same issue in our application

ioquatix commented:
> I think you need to modify https://github.com/macournoyer/thin/blob/master/lib/thin/server.rb and add require_relative "logging".

CI appears fixed.

IkanPakUs commented:
Hi, any idea when these changes will be merged? got same issue in our app
